frontend/javascripts/admin/tasktype/task_type_create_view.tsx

```diff
@@ -43,464 +62,448 @@ function isValidMagnification(_rule, value) {
-    formValues.settings.resolutionRestrictionsForm = {
-      shouldRestrict:
-        // @ts-expect-error ts-migrate(2339) FIXME: Property 'resolutionRestrictions' does not exist o... Remove this comment to see the full error message
-        formValues.settings.resolutionRestrictions.min != null ||
-        // @ts-expect-error ts-migrate(2339) FIXME: Property 'resolutionRestrictions' does not exist o... Remove this comment to see the full error message
-        formValues.settings.resolutionRestrictions.max != null,
-      // @ts-expect-error ts-migrate(2339) FIXME: Property 'resolutionRestrictions' does not exist o... Remove this comment to see the full error message
-      min: formValues.settings.resolutionRestrictions.min || 1,
-      // @ts-expect-error ts-migrate(2339) FIXME: Property 'resolutionRestrictions' does not exist o... Remove this comment to see the full error message
-      max: formValues.settings.resolutionRestrictions.max || 512,
-    };
-    // @ts-expect-error ts-migrate(2339) FIXME: Property 'resolutionRestrictions' does not exist o... Remove this comment to see the full error message
-    delete formValues.settings.resolutionRestrictions;
-    const form = this.formRef.current;
-
-    if (!form) {
-      Toast.info(messages["ui.no_form_active"]);
-      return;
-    }
-
-    form.setFieldsValue(formValues);
-
-    if (taskType?.recommendedConfiguration != null) {
-      // Only "activate" the recommended configuration checkbox if the existing task type contained one
-      this.setState({
-        useRecommendedConfiguration: true,
-      });
+    const defaultFormValues: Partial<FormValues> = _.merge({}, defaultValues, taskType);
+    form.setFieldsValue(defaultFormValues);
+
+    if (taskType?.recommendedConfiguration) {
+      //  Only "activate" the recommended configuration checkbox if the existing task type contained one
+      setUseRecommendedConfiguration(true);
+      form.setFieldValue(
+        ["recommendedConfiguration"],
+        jsonStringify(taskType?.recommendedConfiguration),
+      );
     }
-  }
 
-  async fetchData() {
-    this.setState({
-      isFetchingData: true,
-    });
-    const editableTeams = await getEditableTeams();
-    this.setState({
-      teams: editableTeams,
-      isFetchingData: false,
-    });
+    if (taskType?.settings.resolutionRestrictions)
+      form.setFieldValue(["isResolutionRestricted"], true);
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'formValues' implicitly has an 'any' typ... Remove this comment to see the full error message
-  onFinish = async (formValues) => {
-    if (!this.state.useRecommendedConfiguration) {
-      formValues.recommendedConfiguration = null;
-    }
-
-    const { recommendedConfiguration, settings: rawSettings, ...rest } = formValues;
-    const settings = getMagnificationAdaptedSettings(rawSettings);
```

Comment:
Since the interface of `settings.resolutinsRestriction` is identical between the form and a task tpye in the new code, this method is no longer required.